kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/kafka/validators/SchemaValidatorTest.java

```diff
@@ -0,0 +1,143 @@
+package uk.gov.homeoffice.digital.sas.kafka.validators;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.system.CapturedOutput;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.homeoffice.digital.sas.config.TestConfig;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static uk.gov.homeoffice.digital.sas.Constants.TestConstants.KAFKA_INVALID_SCHEMA_FORMAT_JSON_MESSAGE;
+import static uk.gov.homeoffice.digital.sas.Constants.TestConstants.KAFKA_INVALID_VERSION;
+import static uk.gov.homeoffice.digital.sas.Constants.TestConstants.KAFKA_JSON_MESSAGE;
+import static uk.gov.homeoffice.digital.sas.Constants.TestConstants.KAFKA_VALID_RESOURCE;
+import static uk.gov.homeoffice.digital.sas.Constants.TestConstants.KAFKA_VALID_VERSION;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INCORRECT_FORMAT;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID_VERSION;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_VALIDATED;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.SCHEMA_COMMA_DELIMETER;
+import com.vdurmont.semver4j.Semver;
+
+@SpringBootTest(classes = TestConfig.class)
+@ExtendWith({OutputCaptureExtension.class})
+class SchemaValidatorTest {
+
+  SchemaValidator schemaValidator;
+
+  @BeforeEach
+  void setup () {
+    schemaValidator = new KafkaSchemaValidatorImpl();
+  }
+
+  @Test
+  void should_returnTrue_when_schemaValid() {
+    // given
+    ReflectionTestUtils.setField(schemaValidator, "supportedVersion", "^0.x.x");
```

Comment:
If you make the constructor injection change mentioned in the validator, then we can avoid using these reflection util methods and just pass the supported version in to the constructor.
Same would apply to the other tests here.